CodeGeneration/Sources/SyntaxSupport/Child.swift

```diff
@@ -214,8 +214,7 @@ public class Child {
     documentation: String? = nil,
     isOptional: Bool = false
   ) {
-    precondition(name.first?.isUppercase ?? true, "The first letter of a child’s name should be uppercase")
-    precondition(deprecatedName?.first?.isUppercase ?? true, "The first letter of a child’s name should be uppercase")
+    precondition(name.first?.isLowercase ?? true, "The first letter of a child’s name should be lowercase")
```

Comment:
I think we should also precondition that `deprecatedName` starts with a lowercase letter.

Reply:
I agree with you, but before selecting the exact implementation, I would like to discuss the best option. This is why I've added [this comment](https://github.com/apple/swift-syntax/pull/2055#discussion_r1292200947):

> I'm uncertain about the optimal way to handle this edge case.
>
> If we want to make `deprecatedName` lowercase as well, I'm not sure there's a better solution than simply hardcoding the condition `if deprecatedName == "eofToken" { ... ` somewhere 😕

Sorry if it wasn't understandable in the first place.

So, we are dealing with an edge case where `deprecatedName` is set to `"EOFToken"`, and I'm wondering what would be the best approach to address this situation. The best solution that comes to my mind is hardcoding the condition `if deprecatedName == "eofToken" { ... ` somewhere... :/